index.d.ts

```diff
@@ -0,0 +1,26 @@
+declare module 'node-tunnel' {
+	import * as Promise from 'bluebird';
+	import * as http from 'http';
+	import * as net from 'net';
+
+	export class Request extends http.IncomingMessage {
+		auth?: {
+			username?: string;
+			password?: string;
+		};
+		url: string;
+	}
+
+	export type Middleware = (req: Request, cltSocket: net.Socket, head: Buffer, next: () => void) => void;
+
+	export class Tunnel extends NodeJS.EventEmitter {
+		connect(port: number, host: string, client: net.Socket, req: Request): Promise<net.Socket>;
+		use(middleware: Middleware): void;
+		listen(path: string, callback?: Function): this;
```

Comment:
Should `callback` be typed as something like `(err: void | Error, value?: any) => void`?

Reply:
I copied the typing from @types/node which just used `Function`, I can update it if you would prefer.